x-pack/metricbeat/module/meraki/device_health/device_health.go

```diff
@@ -0,0 +1,260 @@
+		deviceStatuses, err := getDeviceStatuses(m.client, org)
+		if err != nil {
+			return fmt.Errorf("getDeviceStatuses() failed; %w", err)
+		}
+
+		//Get mx device performance score
+		mx_scores, err := getDevicePerformanceScores(m.client, devices)
+		if err != nil {
+			return fmt.Errorf("getDevicePerformanceScores() failed; %w", err)
+		}
+		reportDeviceStatusMetrics(reporter, org, devices, deviceStatuses, mx_scores)
+
+		// //Get &  Report Organization Appliance Uplink
+		appliance_val, appliance_res, appliance_err := m.client.Appliance.GetOrganizationApplianceUplinkStatuses(org, &meraki_api.GetOrganizationApplianceUplinkStatusesQueryParams{})
+		if appliance_err != nil {
+			return fmt.Errorf("Appliance.GetOrganizationApplianceUplinkStatuses failed; [%d] %s. %w", appliance_res.StatusCode(), appliance_res.Body(), appliance_err)
+		}
+		//Get & Report Device Uplink Status
+		lossLatencyuplinks, err := getDeviceUplinkLossLatencyMetrics(m.client, org, m.BaseMetricSet.Module().Config().Period)
+		if err != nil {
+			return fmt.Errorf("getDeviceUplinkMetrics() failed; %w", err)
+		}
+		reportApplianceUplinkStatuses(reporter, org, devices, appliance_val, lossLatencyuplinks)
+
+		//Get & Report Device License State
+		cotermLicenses, perDeviceLicenses, systemsManagerLicense, err := getLicenseStates(m.client, org)
+		if err != nil {
+			return fmt.Errorf("getLicenseStates() failed; %w", err)
+		}
+		reportLicenseMetrics(reporter, org, cotermLicenses, perDeviceLicenses, systemsManagerLicense)
+
+		//Get & Report Org Celluar Uplink Status
+		cullular_val, cullular_res, cullular_err := m.client.CellularGateway.GetOrganizationCellularGatewayUplinkStatuses(org, &meraki_api.GetOrganizationCellularGatewayUplinkStatusesQueryParams{})
+		if cullular_err != nil {
+			return fmt.Errorf("CellularGateway.GetOrganizationCellularGatewayUplinkStatuses failed; [%d] %s. %w", cullular_res.StatusCode(), cullular_res.Body(), cullular_err)
+		}
+		reportCellularGatewayApplianceUplinkStatuses(reporter, org, devices, cullular_val)
+
+		//Get Org Networks
+		//Get Network Health by Org Network
+		//Report NetworkHalthChannelUtilization
+		orgNetworks, orgNetwork_res, orgNetwork_err := m.client.Organizations.GetOrganizationNetworks(org, &meraki_api.GetOrganizationNetworksQueryParams{})
+		if orgNetwork_err != nil {
+			return fmt.Errorf("Organizations.GetOrganizationNetworks failed; [%d] %s. %w", orgNetwork_res.StatusCode(), orgNetwork_res.Body(), orgNetwork_err)
+		}
+		networkHealthUtilizations, err := getNetworkHealthChannelUtilization(m.client, orgNetworks)
+		if err != nil {
+			return err
+		}
+		reportNetworkHealthChannelUtilization(reporter, org, devices, networkHealthUtilizations)
+
+		// Get and Report Organization Wireless Devices Channel Utilization
+		wireless_res, wireless_err := m.client.Devices.GetOrganizationWirelessDevicesChannelUtilizationByDevice(org, &meraki_api.GetOrganizationWirelessDevicesChannelUtilizationByDeviceQueryParams{})
```

Comment:
this is still getting 7 days worth of data every time

Reply:
I believe the wireless devices returns only 1 hr (3600 seconds) by default ... https://developer.cisco.com/meraki/api-v1/get-organization-wireless-devices-channel-utilization-by-device/

Reply:
See my comment on network, should we set that from 1 day to 3600 seconds, so these are both the same ...